src/utils/AdaptiveTimeStep.cpp

```diff
@@ -192,3 +240,42 @@ AdaptiveTimeStep::Calculate (amrex::Real& dt, MultiBeam& beams, amrex::Real plas
         dt = std::min(dt, m_dt_max);
     }
 }
+
+void
+AdaptiveTimeStep::CalculateFromDensity (amrex::Real t, amrex::Real& dt, MultiPlasma& plasmas)
+{
+    HIPACE_PROFILE("AdaptiveTimeStep::Calculate()");
+
+    using namespace amrex::literals;
+
+    if (m_do_adaptive_time_step == 0) return;
+    if (m_adaptive_control_phase_advance == 0) return;
+
+    const PhysConst pc = get_phys_const();
+
+    amrex::Real dt_sub = dt / m_adaptive_phase_substeps;
+    amrex::Real phase_advance = 0.;
+    amrex::Real phase_advance0 = 0.;
+
+    // Get plasma density at beginning of step
+    const amrex::Real plasma_density = plasmas.maxDensity(pc.c * t);
+    const amrex::Real omega_p = std::sqrt(plasma_density * pc.q_e * pc.q_e / (pc.ep0 * pc.m_e));
+    amrex::Real omgb0 = omega_p / std::sqrt(2. *m_min_uz);
```

Comment:
```suggestion
    const amrex::Real omega_p0 = std::sqrt(plasma_density * pc.q_e * pc.q_e / (pc.ep0 * pc.m_e));
    amrex::Real omgb0 = omega_p0 / std::sqrt(2. *m_min_uz);
```

Maybe rename them for consistency, so one cannot confuse it with the `omega_p` in the loop?

Reply:
Indeed, I renamed it `omgp0`, consistent with `omgb0`.